crates/sargon/src/system/sargon_os/sargon_os_delete_account.rs

```diff
@@ -0,0 +1,131 @@
+use crate::prelude::*;
+use radix_engine_interface::blueprints::account::{
+    AccountRemoveAuthorizedDepositorInput as ScryptoAccountRemoveAuthorizedDepositorInput,
+    AccountRemoveResourcePreferenceInput as ScryptoAccountRemoveResourcePreferenceInput,
+};
+use std::future::Future;
+
+// ==================
+// Delete Account (Public)
+// ==================
+impl SargonOS {
+    pub async fn create_delete_account_manifest(
```

Comment:
@matiasbzurovski do you want to handle transfer of resources in a separate PR?

Reply:
I can handle it here as well, it would basically be:
- checking with GW every resource that `account_address` has
- Sending them over to a `recipient_account_address` that will be received by parameter 

right?

Reply:
`create_delete_account_manifest` is untested it seems? Maybe add a simple test?

Reply:
@matiasbzurovski for resource transfer:
- If there is no `recipient_account_address` provided as input, there is no asset transfer.
- If it is provided, then you need to fetch all of the resources for the deleted account - fungibles and non-fungibles. Pagination is required.
- It is needed to check total number of resources, similar to [account_locker claim](https://github.com/radixdlt/sargon/blob/4a5a557b3e373421cf0ed2c26d2c2db51a6d85f0/crates/sargon/src/wrapped_radix_engine_toolkit/high_level/sargon_specific_types/account_locker/account_locker_claimable_resource.rs#L20). If it is greater than 50, let's use the same amount, we should throw a specific error - too many resources.
- If the above check passes, we need to create the manifest similar to [claiming from locker](https://github.com/radixdlt/sargon/blob/4a5a557b3e373421cf0ed2c26d2c2db51a6d85f0/crates/sargon/src/wrapped_radix_engine_toolkit/high_level/manifest_building/manifest_account_locker.rs#L26).